src/mon/MDSMonitor.cc

```diff
@@ -433,7 +433,7 @@ bool MDSMonitor::preprocess_beacon(MonOpRequestRef op)
     } else {
       effective_epoch = fsmap.get_filesystem(fscid)->mds_map.epoch;
     }
-    if (effective_epoch != m->get_last_epoch_seen()) {
+    if ((state != MDSMap::STATE_DNE) && (effective_epoch != m->get_last_epoch_seen())) {
```

Comment:
Let's move this check under line 420. If the requested state is `STATE_DNE`, return false immediately.

Reply:
Fixed it ,thanks!